hugr-core/src/export.rs

```diff
@@ -71,7 +72,10 @@ struct Context<'a> {
     implicit_imports: FxHashMap<&'a str, model::NodeId>,
 
     /// Map from node ids in the [`Hugr`] to the corresponding node ids in the model.
-    node_indices: FxHashMap<Node, model::NodeId>,
+    node_to_id: FxHashMap<Node, model::NodeId>,
+
+    /// Mapping from node ids in the [`Hugr`] to the corresponding model nodes.
+    id_to_node: FxHashMap<model::NodeId, Node>,
```

Comment:
if these are meant to stay in sync would be better to have a dedicated struct holding both that has responsibility for their correctness.

Reply:
Ideally yes. I'm not entirely sure yet if the way that functions are exported is ideal here, so there might be a bit of churn around these things happening. In that light I'm not sure how much it is worth to polish.

Reply:
TODO added.